File: app/src/main/java/com/example/uos_lms/core/domain/model/Assignment.java
package com.example.uos_lms.core.domain.model;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@Getter
@Builder(toBuilder = true)
@EqualsAndHashCode
@ToString
public class Assignment {
    private final String id;
    private final String subjectId;
    private final String departmentId;
    private final String semesterId;
    private final String title;
    private final String description;
    private final long dueDateMillis;
    private final int maxMarks;
    private final String createdBy;
    @Builder.Default
    private final long createdAt = 0L;
    private final String fileUrl;
    private final String fileName;
    private final String filePublicId;
    private final String fileResourceType;
    private final Long fileSize;
}

File: backend/src/models/Assignment.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const assignmentSchema = new Schema(
  {
    subjectId: { type: Schema.Types.ObjectId, ref: 'Subject', required: true },
    departmentId: { type: Schema.Types.ObjectId, ref: 'Department', required: true },
    semesterId: { type: Schema.Types.ObjectId, ref: 'Semester', required: true },
    title: { type: String, required: true, trim: true },
    description: { type: String, default: '' },
    dueDate: { type: Date, required: true },
    maxMarks: { type: Number, required: true, min: 1 },
    createdBy: { type: String, ref: 'User', required: true },
    // Optional Cloudinary attachment via the Media API - null if the teacher didn't attach one.
    // mediaId lets delete cleanly cascade to DELETE /api/media/:id; the rest are denormalized
    // display fields so list screens don't need a join per row.
    mediaId: { type: Schema.Types.ObjectId, ref: 'Media', default: null },
    fileUrl: { type: String, default: null },
    fileName: { type: String, default: null },
    filePublicId: { type: String, default: null },
    fileResourceType: { type: String, default: null },
    fileSize: { type: Number, default: null },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Assignment', assignmentSchema);
